src/backend/libpq/auth.c

```diff
@@ -329,16 +330,15 @@ auth_failed(Port *port, int status, char *logdetail)
 			errstr = gettext_noop("authentication failed for user \"%s\": invalid authentication method");
 			break;
 	  }
+	  cdetail = psprintf(_("Connection matched pg_hba.conf line %d: \"%s\""),
```

Comment:
Why you remove cdetail for case when port->hba == NULL ?

Reply:
If port->hba == NULL, we will be unable to access port->hba->linenumber or port->hba->rawline.